frontend/catalyst/utils/jax_cpu_lapack_kernels/lapack_kernels_using_lapack.cpp

```diff
@@ -35,126 +35,139 @@
 
 #include "lapack_kernels.hpp"
 
+// With SciPy OpenBLAS, symbols are now prefixed with scipy_ when using scipy>=1.14 or
+// scipy-openblas32, whereas this is not the case for the reference implementation used on M1 mac.
+#if defined(__APPLE__) && defined(__arm64__)
+#define SYM_PREFIX
+#else
+#define SYM_PREFIX scipy_
+#endif
+
+// No idea why 3 layers are required here.
+#define CONCAT2(X, Y) X##Y
+#define CONCAT(X, Y) CONCAT2(X, Y)
+#define GET_SYMBOL(X) CONCAT(SYM_PREFIX, X)
```

Comment:
```suggestion
// The CONCAT macro and its helper CONCAT_ are required here since macro arguments are not expanded
// around a ## preprocessing token. See http://port70.net/%7Ensz/c/c11/n1570.html#6.10.3.1.
#define CONCAT_(X, Y) X##Y
#define CONCAT(X, Y) CONCAT_(X, Y)
#define GET_SYMBOL(X) CONCAT(SYM_PREFIX, X)
```